serial/serialposix.py

```diff
@@ -748,6 +748,27 @@ def send_break(self, duration=0.25):
             raise PortNotOpenError()
         termios.tcsendbreak(self.fd, int(duration / 0.25))
 
+    def _update_rts_and_dtr_state(self):
+        """Set terminal status line: Request To Send and Data Terminal Ready"""
+        if self._dtr_state:
+            if self._rts_state:
+                fcntl.ioctl(self.fd, TIOCMBIS, TIOCM_DTRRTS_str)
+            else:
+                s = fcntl.ioctl(self.fd, TIOCMGET, TIOCM_all_str)
+                v = struct.unpack('I', s)[0]
+                v |= TIOCM_DTR
+                v &= ~TIOCM_RTS
+                fcntl.ioctl(self.fd, TIOCMSET, struct.pack('I', v))
+        else:
+            if self._rts_state:
+                s = fcntl.ioctl(self.fd, TIOCMGET, TIOCM_all_str)
+                v = struct.unpack('I', s)[0]
+                v &= ~TIOCM_DTR
+                v |= TIOCM_RTS
+                fcntl.ioctl(self.fd, TIOCMSET, struct.pack('I', v))
+            else:
+                fcntl.ioctl(self.fd, TIOCMBIC, TIOCM_DTRRTS_str)
```

Comment:
I think this formulation is more readable, but please verify that it's actually equivalent! What are your thoughts?


```suggestion
        s = fcntl.ioctl(self.fd, TIOCMGET, TIOCM_all_str)
        v = struct.unpack('I', s)[0]

        # Set or clear DTR.
        if self._dtr_state:
            v |= TIOCM_DTR
        else:
            v &= ~TIOCM_DTR

        # Set or clear RTS.
        if self._rts_state:
            v |= TIOCM_RTS
        else:
            v &= ~TIOCM_RTS

        fcntl.ioctl(self.fd, TIOCMSET, struct.pack('I', v))
```